src/common/src/collection/estimate_size.rs

```diff
@@ -37,27 +40,63 @@ impl EstimateSize for FixedBitSet {
     }
 }
 
-impl EstimateSize for Vec<u8> {
+impl EstimateSize for String {
     fn estimated_heap_size(&self) -> usize {
         self.capacity()
     }
 }
 
-impl<T> EstimateSize for Vec<T>
-where
-    T: EstimateSize,
-{
+impl EstimateSize for () {
+    fn estimated_heap_size(&self) -> usize {
+        0
+    }
+}
+impl<T: EstimateSize> EstimateSize for Box<T> {
     fn estimated_heap_size(&self) -> usize {
-        self.capacity() * std::mem::size_of::<T>()
-            + self
-                .iter()
-                .map(EstimateSize::estimated_heap_size)
-                .sum::<usize>()
+        self.as_ref().estimated_size()
     }
 }
 
-impl EstimateSize for String {
+impl<T: EstimateSize> EstimateSize for Option<T> {
     fn estimated_heap_size(&self) -> usize {
-        self.capacity()
+        if let Some(inner) = self {
+            inner.estimated_heap_size()
+        } else {
+            0
+        }
+    }
+}
+
+/// SAFETY: `Bytes` can store a pointer in some cases, that may cause the size
+/// of a `Bytes` be calculated more than one and when memory stats is larger than the real value.
+impl EstimateSize for Bytes {
+    fn estimated_heap_size(&self) -> usize {
+        self.len()
```

Comment:
Does it has sth similar to `capacity`?

Reply:
No. You can think it as len is always that same as capacity